### PR TITLE
Introduced a JSON schema of a leapp report

### PR DESCRIPTION
**Pull request by fernflower**
Generated by https://jsonschema.net and adjusted manually.

OAMG-4263

### CODE REVIEW

**Comment by pirat89** [comment]
@fernflower  We should refer to the schema from the upstream documentation.

**Comment by fernflower** [comment]
@pirat89 I will put the reference once the schema is merged. Are you ok with the placing in leapp's root?

**Comment by pirat89** [comment]
@fernflower I am thinking where to put it as we want to reduce the main menu. We should discuss it in future with @mportman12  probably. But temporary it could be ok. Btw, why to do it separately? Why not in this PR? As these two things are two sides of the same coin. I guess it's ok unless the jira task is kept in development or if you create the followup task, so we do not foret to finish it.

**Comment by pirat89** [comment]
@fernflower aah...I thought you are talking about readthedocs menu. Now I see where the file is located :) I think it should be in a subdir. As well, haven't we talked about version schema? As we plan to change it in future, so I believe there should be something like version 1.0 .. maybe the file should reflect that as well and should be called `report-schema-v1.0.json`. wdyt?
I am not sure which directory to use to put the schema inside. @vinzenz any idea?

**Comment by pirat89** [comment]
leapp-ci build

**Comment by fernflower** [comment]
@pirat89 what about putting 'version' in the schema itself (did it in the latest commit)? I'd rather keep the filename as it is `report-schema.json` tbh, if we follow the 'do not remove any fields from json' concept the latest file should be supported anyway.

**Comment by pirat89** [comment]
@fernflower  And how you will point people to various versions of schemas then? we will need keep several versions of schemas documented in the same time..

**Comment by pirat89** [comment]
I really believe we need to have info about the version inside the schema and reflect that in the filename as well.

**Review by pirat89 — APPROVED** [review]
giving my approval to not block the pr. I will try to think where to move the file. in the worst case, we can still move it after the PR merge.

**Comment by pirat89** [comment]
replaced tabs by whitespaces in the fixup commits

File: report-schema-v100.json
Comment (pirat89) [inline report-schema-v100.json:326]: What about to add the `uri` format to make validation even more strict?"

```suggestion
                                                            "type": "string",
                                                            "format": "uri",
```

ref: https://json-schema.org/understanding-json-schema/reference/string.html#resource-identifiers

File: report-schema-v100.json
Comment (pirat89) [inline report-schema-v100.json:446]: properties cannot contain "anyOf". this error is present in both files

**Comment by pirat89** [comment]
Fixing the indentation as it was broken on several places.

File: report-schema-v100.json
Comment (pirat89) [inline report-schema-v100.json:492]: This is wrong actually. E.g. 
```
{
  "type": "command",
  "context": [
    "leapp", "answer", "--section", "remove_pam_pkcs11_module_check.confirm=True"
  ]
}

```

We should add more examples to be able to validate the json schema better. Let's do it the next morning.

File: report-schema-v100.json
Comment (pirat89) [inline report-schema-v100.json:449]: and this is probably not possible as well. properties should be always key-value dicts as it looks like.

**Review by pirat89 — DISMISSED** [review]

**Review by pirat89 — APPROVED** [review]
:partying_face:

**Pull request merged**
